src/event.bus/connection.ts

```diff
@@ -126,10 +126,14 @@ export class Connection {
           if (!msg) {
             return logger.error('Invalid incoming message');
           }
-          const { spaceId, purpose, personaServiceId } = JSON.parse(
-            JSON.parse(msg.content.toString())
+          const { bodyOfKnowledgeId, type, purpose, personaServiceId } =
+            JSON.parse(JSON.parse(msg.content.toString()));
```

Comment:
why the double parse?

Reply:
Some quirk of the armqp client we use...